### DISCUSSION TITLE
Why does cost increase with the number of iterations

### DISCUSSION THREAD

**Discussion by lilintong22222** [osqp/osqp#552] [General]
I save the pri_res, eps_prim, dua_res, eps_dual, constraint ( l <= Ax + 1e-3 && Ax - 1e-3 <= u), cost of each iteration, and then plot the curve:
![image](https://user-images.githubusercontent.com/7465438/109159158-470dc900-77af-11eb-9744-b462354e7668.png)

Why does cost increase with the number of iterations ?
I check P matrix is positive definite

**Comment by goulart-paul** [comment]
It does not appear that the solver has converged based on the profile of the primal residual.    I don't think you can infer much from the computed cost at a non-optimal and non-feasible point.    Are you sure that your problem is feasible?

**Comment by lilintong22222** [comment]
1. I check `P` is a positive definite matrix
2. as can be seen from the figure above, when iter is between 50 and 200, the result `x` meets the constraint conditions.  It shows that the constraint has a feasible region ?

Are there any other conditions that need to be met

**Comment by goulart-paul** [comment]
It evidently has not reached an optimal point, otherwise the solver would have stopped.   The primal residual still appears to be above 0.001.   Depending on your stopping criteria, that may not be sufficient.   Your plot appears to show that the `eps_prim` setting is below that.

The conditions that need to be met before the solver stops are explained clearly in the solver documentation here:  https://osqp.org/docs/solver/index.html#convergence
